internal/pkg/cli/pipeline_update.go

```diff
@@ -169,6 +166,19 @@ func (o *updatePipelineOpts) deployPipeline(in *deploy.CreatePipelineInput) erro
 	}
 	if !exist {
 		o.prog.Start(fmt.Sprintf(fmtPipelineUpdateStart, color.HighlightUserInput(o.pipelineName)))
+
+		// If the source requires CodeStar Connections, the user is prompted to update the connection status.
+		if o.shouldPromptUpdateConnection {
+			repo := in.Source.(*deploy.BitbucketSource)
+			connectionName, err := repo.ConnectionName()
```

Comment:
nit: to future proof this:
```go
type codestar interface {
   ConnectionName() (string, error)
}
source, ok := in.Source.(codestar)
if !ok {
  return fmt.Errorf("unexpected error source %v does not have a connection name, in.Source)
}
... // now you can do source.ConnectionName()
```
The advantage of the above is that we won't have to change this block of code when we add GitHub and GitHubEnterprise as types that implement the `ConnectionName` method.